packages/sitecore-jss/src/utils/env.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Parses raw env variable value into a Record
+ * @param {string} mvEnv input
+ * @returns non-empty Record
+ * @throws error when input JSON can't be parsed
+ */
+export const parseMultiValueAsRecord = <T>(mvEnv: string): T => {
+  try {
+    return JSON.parse(mvEnv) as T;
+  } catch (error) {
+    console.log('Parsing of multivalue env variable failed');
+    console.log(`Attempted to parse ${mvEnv}`);
+    throw error;
+  }
+};
+
+/**
+ * Method to parse env variables
+ * @param {string} envValue - can be undefined when providing values via process.env
+ * @param {T} defaultValue - default value
+ * @returns {T | string} parsed value, returns envValue if it's not an object
+ */
+export const tryParseEnvValue = <T>(envValue: string | undefined, defaultValue: T): T | string => {
```

Comment:
Seems POS case-specific still having string in the return type. Sort of defeats the purpose of defaultValue. If we can't parse the value to `T` for any reason then it should return the defaultValue.

Combining these 2 methods (see previous comment), something like this:
```
/**
 * Method to parse JSON-formatted environment variables
 * @param {string | undefined} envValue - can be undefined when providing values via process.env
 * @param {T} defaultValue - default value
 * @returns {T} parsed value
 */
export const tryParseEnvValue = <T>(envValue: string | undefined, defaultValue: T): T => {
  if (!envValue) {
    return defaultValue;
  }
  if (envValue.startsWith('{') && envValue.endsWith('}')) {
    try {
      return JSON.parse(envValue) as T;
    } catch (error) {
      console.warn('Parsing of env variable failed');
      console.warn(`Attempted to parse ${envValue}`);
      return defaultValue;
    }
  }
  return defaultValue;
};
```

Reply:
I like this approach